src/p2p/net_node.inl

```diff
@@ -2227,6 +2227,13 @@ namespace nodetool
 
     network_zone& zone = m_network_zones.at(context.m_remote_address.get_zone());
 
+    if(arg.node_data.peer_id == zone.m_config.m_peer_id)
```

Comment:
I think its worth dropping a short comment about metadata association here, so its clear to future maintainers.

Reply:
done, but I think the attack also works in the other direction, though the attacker has to wait till you connect. Alice connects to Eve, Eve replies with a peer ID she wants to check if it's Alice's on another network type.